backend.native/tests/runtime/concurrent/shared0.kt

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package runtime.concurrent.shared0
+
+import kotlin.test.*
+
+import kotlin.native.concurrent.*
+import kotlin.native.internal.GC
+import kotlin.native.ref.WeakReference
+
+class A(var a: Int)
+
+val global1: SharedReference<A> = SharedReference(A(3))
+
+@Test
+fun testGlobal() {
+    assertEquals(3, global1.value.a)
+
+    val worker = Worker.start()
+    val future = worker.execute(TransferMode.SAFE, {}) {
+        global1
+    }
+
+    val value = future.result
+    assertEquals(3, value.value.a)
+    worker.requestTermination().result
+}
+
+val global2: SharedReference<A> = SharedReference(A(3))
+
+@Test
+fun testGlobalDenyAccessOnWorker() {
+    assertEquals(3, global2.value.a)
+
+    val worker = Worker.start()
+    val future = worker.execute(TransferMode.SAFE, {}) {
+        assertFailsWith<IncorrectDereferenceException> {
+            global2.value
```

Comment:
Accessing to top-level fields in Kotlin/Native can fail with `IncorrectDereferenceException` too. So it makes a lot of sense to move `global2` out of this `assertFailsWith` too.